src/EasyAbp.BookingService.Domain/EasyAbp/BookingService/AssetOccupancyProviders/AssetOccupancyProviderBase.cs

```diff
@@ -281,80 +287,140 @@ public virtual async Task<CanOccupyResult> CanOccupyByCategoryAsync(OccupyAssetB
         return cachedPeriods[(asset.Id, date)];
     }
 
-    [UnitOfWork(true)]
     public virtual async Task<(ProviderAssetOccupancyModel, AssetOccupancy)> OccupyAsync(OccupyAssetInfoModel model,
         Guid? occupierUserId)
     {
-        // Todo: lock the day?
+        using var uow = UnitOfWorkManager.Begin(true, true);
```

Comment:
I don't think a UOW should be created here. Think about the Orleans provider.

Reply:
Avoid using `requiresNew` if possible.

Reply:
> Avoid using `requiresNew` if possible.

The global lock shoule be released after data is saved, so we need a new `UnitOfWork` instead of `ChildUnitOfWork`